pkl-gradle/pkl-gradle.gradle.kts

```diff
@@ -71,3 +73,14 @@ signing {
     }
   }
 }
+
+buildConfig {
+  sourceSets {
+    named("test") {
+      packageName("org.pkl.gradle.constants.test")
+      useKotlinOutput { topLevelConstants = true }
+
+      buildConfigField("KOTLIN_VERSION", libs.versions.kotlin)
+    }
+  }
+}
```

Comment:
The Gradle tests now pin to the same Kotlin version as the build. A symbol, `KOTLIN_VERSION`, is generated within the tests, which is used in the embedded Gradle build.